doc/developer-guides/asm_coding_guidelines.rst

```diff
@@ -704,6 +704,41 @@ Compliant example::
        /* This is a comment. This is a comment. This is a comment. This is a comment. This is a comment. This is a comment. This is a comment. */
 
 
+ASM-CS-08: Legal entity shall be documented in every file
+=========================================================
+
+Legal entity shall be documented in a separate comments block at the start of
+every file.
+The following information shall be included:
+
+a) Copyright
+b) License
+
+Compliant example::
+
+    /* Legal entity shall be placed at the start of the file. */
+    -------------File Contents Start After This Line------------
+    
+    /*
+     * Copyright (C) 2019 Intel Corporation. All rights reserved.
```

Comment:
"All rights reserved." is not compatible with the BSD-3-Clause license, so don't include that sentence.